src/coreclr/gc/gc.cpp

```diff
@@ -10546,19 +10547,52 @@ int gc_heap::object_gennum_plan (uint8_t* o)
 #endif //_MSC_VER && TARGET_X86
 
 #ifdef USE_REGIONS
+void get_initial_region(int nhp, int gen, int hn, uint8_t** region_start, uint8_t** region_end)
+{
+    int initial_region_index;
+    if (gen == poh_generation)
+    {
+        initial_region_index = hn;
+    }
+    else
+    {
+        int soh_start = nhp;
+        if (gen == loh_generation)
+        {
+            int loh_start = soh_start + nhp * (max_generation + 1);
+            initial_region_index = loh_start + hn;
+        }
+        else
+        {
+            initial_region_index = soh_start + (max_generation - gen) * nhp + hn;
+        }
+    }
+    *region_start = initial_regions[2 *initial_region_index];
+    *region_end = initial_regions[2 * initial_region_index + 1];
+}
+
 bool gc_heap::initial_make_soh_regions (gc_heap* hp)
 {
     uint8_t* region_start;
     uint8_t* region_end;
+    uint32_t num_heaps = 1;
+    uint32_t the_heap_number = 0;
```

Comment:
nit, would suggest to not use such verbose var name like `the_heap_number`... you could do what the rest of the code usually does like hn. it's pretty clear what this var means and it's used so locally so a very verbose name seems unnecessary.